server/game/gameSystems/PlayCardSystem.ts

```diff
@@ -18,7 +22,10 @@ export interface IPlayCardProperties extends ICardTargetSystemProperties {
     optional?: boolean;
     entersReady?: boolean;
     playType?: PlayType;
-    // TODO: implement a "nested" property that controls whether triggered abilities triggered by playing the card resolve after that card play or after the whole ability
+    costAdjusterProperties?: ICostAdjusterProperties;
```

Comment:
Ideally I would like to use the same methods that exist for creating cost adjusters, that's the idea that I started with. I think the problem is that those methods are all about creating ongoing effects with cost adjusters, not directly making cost adjusters. It might be possible to use those methods(so PlayCardSystem would accept an IOngoingEffectGenerator technically because that's what they return), but extract the cost adjuster from the generated ongoing effect(without actually applying the effect to the game at all) and pass that to the places that need it. What do you think?

Reply:
Oh yeah I don't think we can directly use the methods for the reason you say, it would be more about using the same naming convention and expected set of properties

Reply:
Alright, I changed PlayCardSystem to directly take a CostAdjuster, and created a CostAdjusterLibrary which I included in AbilityHelper.